src/autoscaling/autoscaled_pool.js

```diff
@@ -15,7 +15,7 @@ const DEFAULT_OPTIONS = {
     loggingIntervalSecs: 60,
     autoscaleIntervalSecs: 10,
 };
-
+const prefixed = createPrefixedNamespace('AutoscaledPool');
```

Comment:
I am thinking if it would not be better to make this a property of an `AutoscaledPool` instance and allow to pass there prefix from above and compose it with this one, i.e. `PuppeteerPool > AutoscaledPool` to have logs in the form:

```
PuppeteerCrawler:AutoscaledPool:setMaxConcurrency() is deprecated, use the "maxConcurrency" property instead
```

possibly you could set up

Reply:
As it would make more clear from where the exception comes from. But maybe it's too long then.

Reply:
I like it, but agree with the length issue. Maybe let's try it and see how it looks?

Reply:
In a case of PuppeteerCrawler/CheerioCrawler vs BasicCrawler - I'd prefix it always with the actual top-level crawler, i.e. PuppeteerCrawler/CheerioCrawler but in a case of AutoScaledPool, I think the two-level namespace does make sense as it helps to find a right place in docs.

Reply:
How about simply having `BasicCrawler.log` field that you can override in `PuppeteerCrawler` ?

Reply:
I think the most consistent way, given how crawlers work together now, would be to pass the logger instance to `BasicCrawler` via constructor options. We could probably keep that option undocumented.

Reply:
the issue is that there are some functions outside of the class that uses logging, that's the reason I'm using a top level variable. so, to not introduce a new variable (like a top `prefix`), I created that function that returns an object. I thought the `prefixed.log` sounded pretty declarative, although a lengthy one. 

but I'll change it

Reply:
@pocesar This only applies to `BasicCrawler` and yeah, you can use top level variables. It's preferable. In `BasicCrawler` there are the `DEFAULT_OPTIONS`. This is just a relict waiting to be refactored. Please move the defaults inside the class, like in `Cheerio` and `Puppeteer` crawlers.